examples/reader-writer.cc

```diff
@@ -0,0 +1,481 @@
+
+    // Create a primitive node named 'int32_field' with type:INT32, repetition:REQUIRED,
+    // logical type:TIME_MILLIS
+    fields.push_back(PrimitiveNode::Make(
+        "int32_field", Repetition::REQUIRED, Type::INT32, LogicalType::TIME_MILLIS));
+
+    // Create a primitive node named 'int64_field' with type:INT64, repetition:REPEATED
+    fields.push_back(PrimitiveNode::Make(
+        "int64_field", Repetition::REPEATED, Type::INT64, LogicalType::NONE));
+
+    fields.push_back(PrimitiveNode::Make(
+        "int96_field", Repetition::REQUIRED, Type::INT96, LogicalType::NONE));
+
+    fields.push_back(PrimitiveNode::Make(
+        "float_field", Repetition::REQUIRED, Type::FLOAT, LogicalType::NONE));
+
+    fields.push_back(PrimitiveNode::Make(
+        "double_field", Repetition::REQUIRED, Type::DOUBLE, LogicalType::NONE));
+
+    // Create a primitive node named 'ba_field' with type:BYTE_ARRAY, repetition:OPTIONAL
+    fields.push_back(PrimitiveNode::Make(
+        "ba_field", Repetition::OPTIONAL, Type::BYTE_ARRAY, LogicalType::NONE));
+
+    // Create a primitive node named 'flba_field' with type:FIXED_LEN_BYTE_ARRAY,
+    // repetition:REQUIRED, field_length = FIXED_LENGTH
+    fields.push_back(PrimitiveNode::Make("flba_field", Repetition::REQUIRED,
+        Type::FIXED_LEN_BYTE_ARRAY, LogicalType::NONE, FIXED_LENGTH));
+
+    // Create a GroupNode named 'schema' using the primitive nodes defined above
+    // This GroupNode is the root node of the schema tree
+    schema_ = std::static_pointer_cast<GroupNode>(
+        GroupNode::Make("schema", Repetition::REQUIRED, fields));
+  }
+
+  parquet::ParquetFileWriter* GetWriter() {
+    if (!writer_) {
+      // Create a parquet file writer using the schema and local file output stream.
+      writer_ = parquet::ParquetFileWriter::Open(ostream_, schema_);
+    }
+    return writer_.get();
+  }
+
+  void Close() { ostream_->Close(); }
+
+ private:
+  std::shared_ptr<parquet::schema::GroupNode> schema_;
+  std::shared_ptr<parquet::LocalFileOutputStream> ostream_;
+  std::unique_ptr<parquet::ParquetFileWriter> writer_;
+};
+
+int main(int argc, char** argv) {
+  /**********************************************************************************
+                             PARQUET WRITER EXAMPLE
+  **********************************************************************************/
+  // parquet::REQUIRED fields do not need definition and repetition level values
+  // parquet::OPTIONAL fields require only definition level values
+  // parquet::REPEATED fields require both definition and repetition level values
+  try {
+    // Create a ParquetWriter instance
+    auto parquet_writer = std::unique_ptr<ParquetWriter>(new ParquetWriter());
+
+    // Setup the schema
+    parquet_writer->SetupSchema();
+
+    // Get a handle to the ParquetFileWriter
+    parquet::ParquetFileWriter* file_writer = parquet_writer->GetWriter();
```

Comment:
It may be slightly pedantic, but I don't think there's a need for the ParquetWriter class here. Instead:

```c++
std::shared_ptr<OutputStream> out_file = std::make_shared<LocalFileOutputStream>(PARQUET_FILENAME);
std::shared_ptr<schema::GroupNode>> schema = SetupSchema();  // static method
std::shared_ptr<ParquetFileWriter> file_writer = ParquetFileWriter::Open(out_file, schema);
```